src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -1349,6 +1364,145 @@ public static PublishPackageViewModel FromLocalPackage(DynamoViewModel dynamoVie
 
         }
 
+        /// <summary>
+        /// The method is used to create a PublishPackageViewModel from a Package object.
+        /// If retainFolderStructure is set to true, the folder structure of the package will be retained. Else, the default folder structure will be imposed.
+        /// </summary>
+        /// <param name="dynamoViewModel"></param>
+        /// <param name="pkg">The package to be loaded</param>
+        /// <param name="retainFolderStructure">If true, the folder structure of the package will be retained as set by the user</param>
+        /// <returns></returns>
+        internal static PublishPackageViewModel FromLocalPackage(DynamoViewModel dynamoViewModel, Package pkg, bool retainFolderStructure)
```

Comment:
How much of the code in the `FromLocalPackage` overload that you marked obsolete is common to this new function? Can you simply call the old function from this one with a few modifications?

Reply:
Most of the code is common, but it is a public method, so this was done to avoid a breaking change. It wont be possible to reuse the old one like you mentioned.